frontend/public/components/factory/details.tsx

```diff
@@ -164,6 +173,12 @@ export type DetailsPageProps = {
   getResourceStatus?: (resource: K8sResourceKind) => string;
   children?: React.ReactNode;
   customKind?: string;
+  alert?: Alert;
+  loaded?: any;
+  loadError?: string;
+  rule?: Rule;
+  silencesLoaded?: boolean;
+  ele?: JSX.Element;
```

Comment:
@sangupta1 why do you need to change details component thereby also changing PageHeading and HorizontalNav?

Ideally we should not change it for a specific resource as this is a generic component used for other resources as well

Reply:
@divyanshiGupta I needed to change the Details component because the PageHeading for Alert comprises of its Monitoring Resource Icon, Severity and 'blue' Silence Alert Button. However, before i changed the PageHeading component, it comprised of showing the Resource status and a 'white' Actions button. So, in order to adjust the changes required for the heading on ALerts page, i changed it. 
![Screenshot from 2021-04-18 23-33-33](https://user-images.githubusercontent.com/77098226/115155674-9e893080-a09e-11eb-8454-6c72acb9044d.png)
![Screenshot from 2021-04-18 23-34-56](https://user-images.githubusercontent.com/77098226/115155709-b365c400-a09e-11eb-88d9-40503a7c9657.png)